src/lib/account_update.ts

```diff
@@ -1877,7 +1877,11 @@ function signJsonTransaction(
       accountUpdate.authorization.proof === null
     ) {
       zkappCommand = JSON.parse(
-        Ledger.signAccountUpdate(JSON.stringify(zkappCommand), privateKey, i)
+        Ledger.signOtherAccountUpdate(
+          JSON.stringify(zkappCommand),
```

Comment:
Do you need to worry about `stringify` safety wrt circular references?

Reply:
nope, none of these transaction data structures has any circular references (and they are kept really explicitly typed so something like an accidental circular ref in there shouldn't happen)